sale_timesheet_line_exclude/tests/test_sale_timesheet_line_exclude.py

```diff
@@ -0,0 +1,175 @@
+                "invoice_policy": "delivery",
+                "uom_id": self.uom_hour.id,
+                "uom_po_id": self.uom_hour.id,
+                "default_code": "CODE-1",
+                "service_type": "timesheet",
+                "service_tracking": "task_global_project",
+                "project_id": project.id,
+                "taxes_id": False,
+                "property_account_income_id": account.id,
+            }
+        )
+        employee = self.SudoEmployee.create(
+            {"name": "Employee #1", "timesheet_cost": 42}
+        )
+        account_payable = self.SudoAccountAccount.create(
+            {
+                "code": "AP4",
+                "name": "Payable #1",
+                "user_type_id": self.user_type_payable.id,
+                "reconcile": True,
+            }
+        )
+        account_receivable = self.SudoAccountAccount.create(
+            {
+                "code": "AR1",
+                "name": "Receivable #1",
+                "user_type_id": self.user_type_receivable.id,
+                "reconcile": True,
+            }
+        )
+        partner = self.SudoPartner.create(
+            {
+                "name": "Partner #1",
+                "email": "partner1@localhost",
+                "property_account_payable_id": account_payable.id,
+                "property_account_receivable_id": account_receivable.id,
+            }
+        )
+        sale_order = self.SudoSaleOrder.create(
+            {
+                "partner_id": partner.id,
+                "partner_invoice_id": partner.id,
+                "partner_shipping_id": partner.id,
+            }
+        )
+        sale_order_line = self.SudoSaleOrderLine.create(
+            {
+                "order_id": sale_order.id,
+                "name": product.name,
+                "product_id": product.id,
+                "product_uom_qty": 0,
+                "product_uom": self.uom_hour.id,
+                "price_unit": product.list_price,
+            }
+        )
+        sale_order.action_confirm()
+        task = self.SudoProjectTask.search([("sale_line_id", "=", sale_order_line.id)])
+        timesheet1 = self.SudoAccountAnalyticLine.create(
+            {
+                "project_id": task.project_id.id,
+                "task_id": task.id,
+                "name": "Entry #1-1",
+                "unit_amount": 1,
+                "employee_id": employee.id,
+            }
+        )
+        timesheet2 = self.SudoAccountAnalyticLine.create(
+            {
+                "project_id": task.project_id.id,
+                "task_id": task.id,
+                "name": "Entry #1-2",
+                "unit_amount": 1,
+                "employee_id": employee.id,
+                "exclude_from_sale_order": False,
+            }
+        )
+
+        self.assertEqual(timesheet1.timesheet_invoice_type, "billable_time")
+        self.assertEqual(timesheet2.timesheet_invoice_type, "billable_time")
```

Comment:
Travis have a problem here. I do not understand the reason and don't get behind it.

Does anybody have a hin here?